sdk/servicebus/service-bus/src/util/typeGuards.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Helper TypeGuard that checks if something is defined or not.
+ * @param thing - Anything
+ * @internal
+ */
+export function isDefined<T>(thing: T | undefined | null): thing is T {
+  return typeof thing !== "undefined" && thing !== null;
+}
+
+/**
+ * Helper TypeGuard that checks if the input is an object with the specified properties.
+ * @param thing - Anything.
+ * @param properties - The name of the properties that should appear in the object.
+ * @internal
+ */
+export function isObjectWithProperties<Thing extends unknown, PropertyName extends string>(
+  thing: Thing,
+  properties: PropertyName[]
+): thing is Thing & Record<PropertyName, unknown> {
+  if (!isDefined(thing) || typeof thing !== "object") {
+    return false;
+  }
+
+  for (const property of properties) {
+    if (!objectHasProperty(thing, property)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * Helper TypeGuard that checks if the input is an object with the specified property.
+ * @param thing - Any object.
+ * @param property - The name of the property that should appear in the object.
+ * @internal
+ */
+export function objectHasProperty<Thing extends unknown, PropertyName extends string>(
+  thing: Thing,
+  property: PropertyName
+): thing is Thing & Record<PropertyName, unknown> {
+  if (!(property in thing)) {
```

Comment:
@chradek Copied this file from Event Hubs where you added this initially. Do you recall why you dont do the below check here?

```js
if (!isDefined(thing) || typeof thing !== "object") {
    return false;
  }
```

Reply:
We could do that. This function isn't exported and is only called by `isObjectWithProperties`, which does this check on `thing` once. If this was defined in `objectHasProperty`, then we'd do the check for each property we're checking in `thing`.

We might be able to change `Thing extends unknown` to extend an empty object instead in this function. Then we'd get a warning if you passed in null or a scalar type.

Reply:
I removed the export here, but it was being exported in Event Hubs

Reply:
Ah, then that was an oversight on my part! I shouldn't have had that exported (or should have tightened the types so Thing extended an object)